packages/extensionmanager/src/widget.tsx

```diff
@@ -656,8 +619,56 @@ export class ExtensionView extends VDomRenderer<ListModel> {
       />
     );
     const content = [];
+    content.push(
+      <CollapsibleSection
+        key="warning-section"
+        isOpen={!ListModel.isDisclaimed()}
+        disabled={false}
+        header={'Warning'}
+      >
+        <div className="jp-extensionmanager-disclaimer">
+          <div>
+            Extensions installed contain arbitrary code that can execute on your
+            machine that may contain malicious code.
+          </div>
+          <div style={{ paddingTop: 8 }}>
+            I understand extensions contain arbitrary code.
+          </div>
+          <div style={{ paddingTop: 8 }}>
+            {ListModel.isDisclaimed() && (
+              <Button
+                className="jp-extensionmanager-disclaimer-disable"
+                onClick={(e: React.MouseEvent<Element, MouseEvent>) => {
+                  this._settings.set('disclaimed', false).catch(reason => {
+                    console.error(
+                      `Something went wrong when setting disclaimed.\n${reason}`
+                    );
+                  });
+                }}
+              >
+                Disable
+              </Button>
+            )}
+            {!ListModel.isDisclaimed() && (
+              <Button
+                className="jp-extensionmanager-disclaimer-enable"
+                onClick={(e: React.MouseEvent<Element, MouseEvent>) => {
+                  this._forceOpen = true;
```

Comment:
Having this property and using the `componentWillRecieveProps` seems like a code smell with respect to React.

Instead, this really should be refactored to use React state I believe, which would have an `isOpen` property that defaults to the `isDisclaimed` prop....

Gonna merge this in however first, since it does work, then work on a refactor.

Reply:
Yeah I hate using componentWillReceiveProps as it will be remove in the next major react release. I have tried the state way but felt into limitations (hard to describe here in text, but linkde to the given props and the relation with the button being clicked, the code to begin to make it work was even more smelly than this  componentWillReceiveProps).

A more deep refactor would be to use React functions and take time to think about the user cases that this component would need to support.

Reply:
> I have tried the state way but felt into limitations (hard to describe here in text, but linkde to the given props and the relation with the button being clicked, the code to begin to make it work was even more smelly than this componentWillReceiveProps).

Yeah might be something we could explore on a call at a later date! I would love to see where it got ugly.